packages/lwc/types.d.ts

```diff
@@ -220,12 +220,28 @@ declare module 'lwc' {
         consumerDisconnectedCallback?: (consumer: ContextConsumer) => void;
     }
 
-    interface WireAdapterConstructor {
-        new (callback: WireDataCallback): WireAdapter;
+    interface WireAdapterConstructor<T> {
+        new (callback: WireDataCallback): WireAdapter<T>;
         configSchema?: Record<string, WireAdapterSchemaValue>;
         contextSchema?: Record<string, WireAdapterSchemaValue>;
     }
 
     type Contextualizer = (elm: EventTarget, options: ContextProviderOptions) => void;
-    export function createContextProvider(config: WireAdapterConstructor): Contextualizer;
+    export function createContextProvider<T>(config: WireAdapterConstructor<T>): Contextualizer;
+}
+
+declare interface WiredValue<T extends (...args: any) => any> {
+    error?: FetchResponse;
+    data?: Awaited<ReturnType<T>>;
+}
```

Comment:
Wire adapters can return any data shape. All the Lightning Data Service wire adapters produce the same kind of object, with a `data` and an `error` property. But it is not the case for all the wire adapters (eg [`CurrentPageReference`](https://developer.salesforce.com/docs/component-library/bundle/lightning-navigation/documentation)).

The `WiredValue` and `FetchResponse` interfaces are specific to Lightning Data Service and should be removed from this type declaration.

Reply:
Thanks, done.